src/sign_workflow/signer_windows.py

```diff
@@ -21,31 +22,46 @@ class SignerWindows(Signer):
 
     ACCEPTED_FILE_TYPES = [".msi", ".exe", ".dll", ".sys", ".ps1", ".psm1", ".psd1", ".cat", ".zip"]
 
-    def generate_signature_and_verify(self, artifact: str, basepath: Path, signature_type: str) -> None:
-        self.sign(artifact, basepath, signature_type)
+    def generate_signature_and_verify(self, artifact: str, basepath: Path, signature_type: str, overwrite_existing: str) -> None:
+        self.sign(artifact, basepath, signature_type, overwrite_existing)
 
     def is_valid_file_type(self, file_name: str) -> bool:
         return any(
             file_name.endswith(x) for x in SignerWindows.ACCEPTED_FILE_TYPES
         )
 
-    def sign(self, artifact: str, basepath: Path, signature_type: str) -> None:
+    def sign(self, artifact: str, basepath: Path, signature_type: str, overwrite_existing: str) -> None:
         filename = os.path.join(basepath, artifact)
-        signed_prefix = "signed_"
-        signature_file = os.path.join(basepath, signed_prefix + artifact)
-        self.__remove_existing_signature__(signature_file)
-        signing_cmd = [
-            "./opensearch-signer-client",
-            "-i",
-            filename,
-            "-o",
-            signature_file,
-            "-p",
-            "windows",
-        ]
-        self.git_repo.execute(" ".join(signing_cmd))
-        signed_folder = os.path.join(basepath, "signed")
-        if not os.path.exists(signed_folder):
-            os.mkdir(signed_folder)
-        signed_location = os.path.join(signed_folder, artifact)
-        os.rename(signature_file, signed_location)
+        if overwrite_existing == 'True':
+            logging.info('Overwriting the existing file {}'.format(filename))
+            signing_cmd = [
+                "./opensearch-signer-client",
+                "-i",
+                filename,
+                "-o",
+                filename,
+                "-p",
+                "windows"
+                " -r",
+                "True"
+            ]
+            self.git_repo.execute(" ".join(signing_cmd))
+        else:
+            signed_prefix = "signed_"
+            signature_file = os.path.join(basepath, signed_prefix + artifact)
+            self.__remove_existing_signature__(signature_file)
+            signing_cmd = [
+                "./opensearch-signer-client",
+                "-i",
+                filename,
+                "-o",
+                signature_file,
+                "-p",
+                "windows"
+            ]
+            self.git_repo.execute(" ".join(signing_cmd))
+            signed_folder = os.path.join(basepath, "signed")
+            if not os.path.exists(signed_folder):
+                os.mkdir(signed_folder)
+            signed_location = os.path.join(signed_folder, artifact)
+            os.rename(signature_file, signed_location)
```

Comment:
Instead of writing two blocks can we just do something similar to this?
https://github.com/opensearch-project/opensearch-build/pull/1726/files#diff-52d601c8af5bd5f8c66790da6f0e84de498460c18a30989e80634b33aba688c3

Reply:
It has whole another logic of creating a subfolder called `signed` by default. Should we change that behavior too?
I think we should and keep it like pgp signing where the generated signature is written in the same location as the original artifact. Let me know what you think.

Reply:
I just feel that the if statement is just for the `overrite=True or False`, which does not reward such a big behavior change on the matter.

If we can have a specific flag on such behavior changes it would be better.

Not a hard blocker for me now, let me know.

Reply:
The filename of the output file also varies in both statements. Hence keeping it that way. The default behavior is also weird. Not sure if we are actually using it anywhere today but surely would like to change it in future.

Reply:
Adding @zelinh to the conversation to get an idea why the old approach was taken.

Reply:
I see the logic here. If the signer is actually implemented to support doing this, I am fine with this logic. The reason that I created another `signed` subfolder is the signer doesn't support to override before.

Reply:
Got it so now since it supports override we can remove this logic completely right?

Reply:
I refactored the logic to support overwrite as well as just add `signed_` prefix if the overwrite is disabled. Let me know if the overwrite should be enabled for windows by default.
@prudhvigodithi @peterzhuamazon @zelinh